sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/ReplaceNullWithFalseInPredicate.scala

```diff
@@ -56,6 +55,12 @@ object ReplaceNullWithFalseInPredicate extends Rule[LogicalPlan] {
     case d @ DeleteFromTable(_, Some(cond)) => d.copy(condition = Some(replaceNullWithFalse(cond)))
     case u @ UpdateTable(_, _, Some(cond)) => u.copy(condition = Some(replaceNullWithFalse(cond)))
     case p: LogicalPlan => p transformExpressions {
+      // For `EqualNullSafe` with a `TrueLiteral`, whether the other side is null or false has no
+      // difference, as `null <=> true` and `false <=> true` both return false.
+      case EqualNullSafe(left, TrueLiteral) =>
```

Comment:
The change is needed here to not fail the existing tests. When we convert `If` to `EqualNullSafe`, we should make sure the rule `ReplaceNullWithFalseInPredicate` still applies.

Reply:
It would be great if someone can pull this out and open a separated PR/JIRA. Otherwise, I'll do it tomorrow.

Reply:
`NullPropagation` should optimize `EqualNullSafe(null, TrueLiteral)` to false too, isn't?

Reply:
It can, but not with `And`, e.g.  `EqualNullSafe(col && null, true)` => `EqualNullSafe(col && false, true)` => `EqualNullSafe(false, true)` => `false`.

Reply:
Oh I see. Makes sense.